### PR TITLE
[bsc#1092140] enable psp opt-out

### PR DESCRIPTION
**Pull request by MaximilianMeister**
they need to be globally unique

orch error happened when setting psp to false in params.sls

fix#conflicting-sls-ids

Signed-off-by: Maximilian Meister <mmeister@suse.de>

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
> what does setting psp to false do?

it's supposed to do the opt-out of pod security policies

**Comment by MaximilianMeister** [comment]
@jordimassaguerpla it seems that setting `psp` to false makes bootstrap fail (see https://github.com/kubic-project/salt/pull/517) - however we should test the same thing with this patch here to confirm that this is the only issue...

**Comment by MaximilianMeister** [comment]
just for the record, setting `psp: false` was green http://jenkins.caasp.suse.net/blue/organizations/jenkins/salt/detail/PR-516/5/pipeline

i'll drop the commit again

File: salt/kube-apiserver/apiserver.jinja [hunk 1]
```diff
@@ -23,7 +23,11 @@ KUBE_ETCD_SERVERS="--etcd-cafile={{ pillar['ssl']['ca_file'] }} \
 KUBE_SERVICE_ADDRESSES="--service-cluster-ip-range={{ pillar['services_cidr'] }}"
 
 # default admission control policies
+{% if salt.caasp_pillar.get('addons:psp', True) %}
 KUBE_ADMISSION_CONTROL="--admission-control=Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,NodeRestriction,PodSecurityPolicy,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota,DefaultTolerationSeconds"
+{% else %}
+KUBE_ADMISSION_CONTROL="--admission-control=Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,NodeRestriction,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota,DefaultTolerationSeconds"
```
Comment (ereslibre) [inline salt/kube-apiserver/apiserver.jinja]: I think we can better have a real array for this, something like:

```
{% set kube_admission_control = ['Initializers', 'NamespaceLifecycle', 'LimitRanger', 'ServiceAccount', 'NodeRestriction', 'PersistentVolumeLabel', 'DefaultStorageClass', 'ResourceQuota', 'DefaultTolerationSeconds'] %}
{% if salt.caasp_pillar.get('addons:psp', True) %}
  {% do kube_admission_control.append('PodSecurityPolicy') %}
{% endif %}
KUBE_ADMISSION_CONTROL="{{ kube_admission_control|join('.') }}"
```

Comment (MaximilianMeister) [reply salt/kube-apiserver/apiserver.jinja]: i was just pushing an update when you comemnted, tried your suggestion afterwards on https://github.com/qn7o/jinja2-live-parser but it didn't work, i have it slightly different

**Pull request merged**